src/java/main/org/apache/zookeeper/server/quorum/QuorumPeer.java

```diff
@@ -1744,9 +1744,14 @@ public void setAcceptedEpoch(long e) throws IOException {
     public boolean processReconfig(QuorumVerifier qv, Long suggestedLeaderId, Long zxid, boolean restartLE) {
        InetSocketAddress oldClientAddr = getClientAddress();
 
+       boolean isReconfigEnabled = QuorumPeerConfig.isReconfigEnabled();
        // update last committed quorum verifier, write the new config to disk
-       // and restart leader election if config changed
-       QuorumVerifier prevQV = setQuorumVerifier(qv, true);
+       // and restart leader election if config changed. If reconfig feature is
+       // not enabled, just set last committed quorum verifier and bail out.
+       QuorumVerifier prevQV = setQuorumVerifier(qv, isReconfigEnabled);
```

Comment:
The server acts according to the config it has in its quorum verifier, so setting quorum verifier overwrites the config file settings with the leader's settings